### PR TITLE
update mtx v4 commands

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: tests/mtx/dlms/get.day.demand.test.ts [hunk 1]
```diff
@@ -34,8 +33,8 @@ const examples = [
                 month: 2,
                 date: 3
             },
-            energyType: A_PLUS_ENERGY_TYPE,
-            energies: [40301230, undefined, undefined, 2145623]
+            energyType: 1,
```
Comment (DarkPark) [inline tests/mtx/dlms/get.day.demand.test.ts:36]: why remove constant?

File: tests/mtx/dlms/get.energy.current.test.ts [hunk 2]
```diff
@@ -19,8 +18,8 @@ const examples = [
     {
         name: 'energy A+',
         parameters: {
-            energyType: A_PLUS_ENERGY_TYPE,
-            energies: [40301230, undefined, undefined, 2145623]
+            energyType: 1,
```
Comment (DarkPark) [inline tests/mtx/dlms/get.energy.current.test.ts:21]: and here

File: tests/mtx/dlms/get.day.demand.test.ts (hunk 1 above)
Comment (DeadByElpy) [reply tests/mtx/dlms/get.day.demand.test.ts:36]: constants restored as file constants/energyTypes.ts, added examples in TSdoc

File: tests/mtx/dlms/get.energy.current.test.ts (hunk 2 above)
Comment (DeadByElpy) [reply tests/mtx/dlms/get.energy.current.test.ts:21]: constants restored as file constants/energyTypes.ts, added examples in TSdoc

File: tests/mtx/message.test.ts [hunk 3]
```diff
@@ -2106,6 +2106,7 @@ export const checkMessage = ( messageLink, messageParams: IMessage ) => {
         expect(getHexFromBytes([messageData.lrc.actual])).toEqual(getHexFromBytes([lrc]));
         expect(messageData.messageId).toEqual(messageId);
         expect(messageData.accessLevel).toEqual(accessLevel);
+        console.log(messageData.id);
```
Comment (DarkPark) [inline tests/mtx/message.test.ts:2109]:
```suggestion
```

File: tests/mtx/dlms/get.day.demand.test.ts (hunk 1 above)
Comment (DarkPark) [reply tests/mtx/dlms/get.day.demand.test.ts:36]: how about use this file here as well?

File: tests/mtx/dlms/get.energy.current.test.ts (hunk 2 above)
Comment (DarkPark) [reply tests/mtx/dlms/get.energy.current.test.ts:21]: and here

File: src/mtx/utils/getObisByEnergy.ts [hunk 4]
```diff
@@ -1,7 +1,5 @@
-import {A_MINUS_ENERGY_TYPE} from './CommandBinaryBuffer.js';
-
 export default ( energy: number | undefined, tariff: number = 0 ): string => {
-    const obis = energy === A_MINUS_ENERGY_TYPE ? '2.8.x' : '1.8.x';
+    const obis = energy === 2 ? '2.8.x' : '1.8.x';
```
Comment (DarkPark) [inline src/mtx/utils/getObisByEnergy.ts:2]: and here

File: src/mtx/utils/CommandBinaryBuffer.ts [hunk 5]
```diff
@@ -707,7 +707,11 @@ export interface IEnergyPeriod {
 export interface IEnergies extends Array<types.TUint32 | null> {}
 
 export interface IPackedEnergiesWithType {
-    energyType?: TEnergyType,
+    /**
+     * `1` - `A+`, `2` - `A-`
```
Comment (DarkPark) [inline src/mtx/utils/CommandBinaryBuffer.ts:711]: would be nice to have @link to constants/energyTypes.ts

File: src/mtx/message/uplink.ts [hunk 6]
```diff
@@ -92,7 +92,8 @@ toBytesMap[commands.getDayProfile.id] = commands.getDayProfile.toBytes;
 toBytesMap[commands.getDeviceId.id] = commands.getDeviceId.toBytes;
 toBytesMap[commands.getDeviceType.id] = commands.getDeviceType.toBytes;
 toBytesMap[commands.getDisplayParam.id] = commands.getDisplayParam.toBytes;
-toBytesMap[commands.getEnergyCurrent.id] = commands.getEnergyCurrent.toBytes;
+toBytesMap[commands.getEnergy.id] = commands.getEnergy.toBytes;
+toBytesMap[commands.getEnergyExport.id] = commands.getEnergyExport.toBytes;
```
Comment (DarkPark) [inline src/mtx/message/uplink.ts:96]: would be nice to sort

File: src/mtx/commands/downlink/getEnergyExport.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+import CommandBinaryBuffer, {ICommandBinaryBuffer} from '../../utils/CommandBinaryBuffer.js';
+import * as command from '../../utils/command.js';
+import {READ_ONLY} from '../../constants/accessLevels.js';
+
+
+interface IGetEnergyCurrentParameters {
+    /**
+     * `1` - `A+`, `2` - `A-`
+     */
+    energyType?: types.TUint8
```
Comment (DarkPark) [inline src/mtx/commands/downlink/getEnergyExport.ts:36]: how about make a new type `TEnergyType` with good description and use it everywhere?

File: src/mtx/commands/uplink/getEnergyExport.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
+/**
+ * Uplink command to get current active energy (`A-`) by default or selected active energy (`A+` or `A-`) for 4 tariffs (`T1`-`T4`).
+ *
+ * @packageDocumentation
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import * as getEnergyCurrent from 'jooby-codec/mtx/commands/uplink/getEnergyCurrent.js';
```
Comment (DarkPark) [inline src/mtx/commands/uplink/getEnergyExport.ts:8]: wrong example

File: tests/mtx/dlms/get.day.demand.test.ts (hunk 1 above)
Comment (DeadByElpy) [reply tests/mtx/dlms/get.day.demand.test.ts:36]: added

File: tests/mtx/dlms/get.energy.current.test.ts (hunk 2 above)
Comment (DeadByElpy) [reply tests/mtx/dlms/get.energy.current.test.ts:21]: added

File: tests/mtx/message.test.ts (hunk 3 above)
Comment (DeadByElpy) [reply tests/mtx/message.test.ts:2109]: removed

**Review by aleks-and — APPROVED** [review]

File: src/mtx/utils/CommandBinaryBuffer.ts (hunk 5 above)
Comment (DeadByElpy) [reply src/mtx/utils/CommandBinaryBuffer.ts:711]: type added

File: src/mtx/commands/downlink/getEnergyExport.ts (hunk 7 above)
Comment (DeadByElpy) [reply src/mtx/commands/downlink/getEnergyExport.ts:36]: added

File: src/mtx/commands/uplink/getEnergyExport.ts (hunk 8 above)
Comment (DeadByElpy) [reply src/mtx/commands/uplink/getEnergyExport.ts:8]: fixed

File: src/mtx/types.ts [hunk 9]
```diff
@@ -12,3 +14,10 @@ export * from '../types.js';
  * READ_ONLY = 0x03;
  */
 export type TAccessLevel = BrandType<number, '0 | 1 | 2 | 3'>;
+
+/**
+ * `1` - `A+`, `2` - `A-`
+ *
+/** One of the {@link energyTypes | energy types}
```
Comment (DarkPark) [inline src/mtx/types.ts:21]:
```suggestion
 * One of the {@link energyTypes | energy types}
```

Comment (DeadByElpy) [reply src/mtx/types.ts:21]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**